velox/functions/prestosql/tests/RepeatTest.cpp

```diff
@@ -0,0 +1,148 @@
+      const std::vector<VectorPtr>& input,
+      const std::string& expectedError) {
+    VELOX_ASSERT_THROW(
+        evaluate<ArrayVector>("repeat(C0, C1)", makeRowVector(input)),
+        expectedError);
+  }
+};
+} // namespace
+
+TEST_F(RepeatTest, repeatInteger) {
+  auto elementVector =
+      makeNullableFlatVector<int32_t>({10, -20, 30, 40, std::nullopt, 50});
+  auto countVector =
+      makeNullableFlatVector<int32_t>({1, 2, 3, 0, 4, std::nullopt});
+  auto expected = makeNullableArrayVector<int32_t>({
+      {{10}},
+      {{-20, -20}},
+      {{30, 30, 30}},
+      {{}},
+      {{std::nullopt, std::nullopt, std::nullopt, std::nullopt}},
+      std::nullopt,
+  });
+  testRepeat(expected, {elementVector, countVector});
+}
+
+TEST_F(RepeatTest, repeatFloat) {
+  auto elementVector = makeNullableFlatVector<float>(
+      {0.0, -2.0, 3.333333, 4.0004, std::nullopt, 5.12345});
+  auto countVector =
+      makeNullableFlatVector<int32_t>({1, 2, 3, 0, 4, std::nullopt});
+  auto expected = makeNullableArrayVector<float>({
+      {{0.0}},
+      {{-2.0, -2.0}},
+      {{3.333333, 3.333333, 3.333333}},
+      {{}},
+      {{std::nullopt, std::nullopt, std::nullopt, std::nullopt}},
+      std::nullopt,
+  });
+  testRepeat(expected, {elementVector, countVector});
+}
+
+TEST_F(RepeatTest, repeatBoolean) {
+  auto elementVector =
+      makeNullableFlatVector<bool>({true, false, true, false, std::nullopt});
+  auto countVector = makeNullableFlatVector<int32_t>({1, 2, 3, 0, 4});
+  auto expected = makeNullableArrayVector<bool>(
+      {{true},
+       {false, false},
+       {true, true, true},
+       {},
+       {std::nullopt, std::nullopt, std::nullopt, std::nullopt}});
+  testRepeat(expected, {elementVector, countVector});
+}
+
+TEST_F(RepeatTest, repeatString) {
+  using S = StringView;
+
+  auto elementVector = makeNullableFlatVector<StringView>(
+      {S("foo"),
+       S("The quick brown fox jumps over the lazy dog"),
+       S("bar"),
+       S("baz"),
+       std::nullopt,
+       S("qux")});
+  auto countVector =
+      makeNullableFlatVector<int32_t>({1, 2, 3, 0, 4, std::nullopt});
+  auto expected = makeNullableArrayVector<StringView>({
+      {{S("foo")}},
+      {{S("The quick brown fox jumps over the lazy dog"),
+        S("The quick brown fox jumps over the lazy dog")}},
+      {{S("bar"), S("bar"), S("bar")}},
+      {{}},
+      {{std::nullopt, std::nullopt, std::nullopt, std::nullopt}},
+      std::nullopt,
+  });
+  testRepeat(expected, {elementVector, countVector});
+}
+
+TEST_F(RepeatTest, tryRepeat) {
+  auto elementVector =
+      makeNullableFlatVector<int32_t>({10, -20, 30, 40, std::nullopt, 50});
+  auto countVector =
+      makeNullableFlatVector<int32_t>({0, 1, -2, 10001, 99999, 3});
+  auto expected = makeNullableArrayVector<int32_t>({
+      {{}},
+      {{-20}},
+      std::nullopt,
+      std::nullopt,
+      std::nullopt,
+      {{50, 50, 50}},
+  });
+  testTryRepeat(expected, {elementVector, countVector});
```

Comment:
Would you also check that without TRY the function throws an error (and make sure the error message is correct)?

Reply:
This is covered in `repeatError`.

Reply:
Hmm... but why split this verification into two tests?

Reply:
So I re-structured the code into two tests:
- test with valid arguments.
- test with an invalid count argument.

Both tests cover w/ and w/o TRY(), constant/non-constant count argument, `null` count argument. Please let me know if I'm missing anything.